LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/NetworkServerDiscovery/DiscoveryService.cs

```diff
@@ -69,7 +70,13 @@ public async Task HandleDiscoveryRequestAsync(HttpContext httpContext, Cancellat
                                               Id6.FormatOptions.FixedWidth);
 
                         var lnsUri = await this.lnsDiscovery.ResolveLnsAsync(stationEui, cancellationToken);
-                        var url = new Uri($"{lnsUri.Scheme}://{lnsUri.Host}:{lnsUri.Port}{DataEndpointPath}/{stationEui}");
+
+                        // Ensure resilience against duplicate specification of `router-data` and make sure that LNS host address ends with slash
+                        // to make sure that URI composes as expected.
+                        var lnsUriSanitized = Regex.Replace(lnsUri.AbsoluteUri, @"/router-data/?", string.Empty, RegexOptions.IgnoreCase);
```

Comment:
Are you sure about `RegexOptions.IgnoreCase`? URL paths are _case-sensitive_.

Reply:
I agree that [the specification](https://www.rfc-editor.org/rfc/rfc7230) says that URL paths are case-sensitive. In practice, some web servers apply case-insensitive path matching (i.e. IIS). What made me choose case-insensitive over case-sensitive matching in this case is Postel's law - the intention of the configuration is clear, we should accept it and avoid adding the `router-data` path twice.

Reply:
> In practice, some web servers apply case-insensitive path matching (i.e. IIS).

I doubted that IIS doesn't respect the rule. I think the misconception might be coming from the fact that the Windows file system uses case-insensitive file names so when serving static files, a URL identifying a file will serve the file if the path is simply different in case. That said, I agree that we can go Postel's way. I'd suggest to add `RegexOptions.CultureInvariant` as well though.

Reply:
Makes sense! I added this in e639f929e5d1c5a90c365d52bdc2fb5d6956b474